### PR TITLE
Fix Docker image builds

### PR DESCRIPTION
**Pull request by joshessman-llnl**
This PR removes some duplicated logic in the Github actions file.

Also, the switch to sundials-by-default and uberenv-as-a-submodule required a few changes to the Dockerfiles.

These modified Dockerfiles were built successfully in https://github.com/LLNL/serac/actions/runs/496550239

### CODE REVIEW

File: host-configs/docker/docker-linux-ubuntu18.04-x86_64-clang@10.0.0.cmake [hunk 1]
```diff
@@ -41,25 +41,25 @@ set(MPIEXEC_EXECUTABLE "/usr/bin/mpiexec" CACHE PATH "")
 #---------------------------------------
 # Library Dependencies
 #---------------------------------------
-set(TPL_ROOT "/home/serac/serac_tpls/clang-10.0.0" CACHE PATH "")
+set(TPL_ROOT "/home/serac/serac_tpls/spack/opt/spack/linux-ubuntu18.04-ivybridge/clang-10.0.0" CACHE PATH "")
 
-set(AXOM_DIR "${TPL_ROOT}/axom-0.4.0serac" CACHE PATH "")
+set(AXOM_DIR "${TPL_ROOT}/axom-0.4.0serac-eawir7ejjglhdcxvjqh7y5rdidn77jco" CACHE PATH "")
```
Comment (jamiebramwell) [inline host-configs/docker/docker-linux-ubuntu18.04-x86_64-clang@10.0.0.cmake]: I thought we got rid of the spack hashes. How did they sneak back in?

**Review by jamiebramwell — APPROVED** [review]

File: host-configs/docker/docker-linux-ubuntu18.04-x86_64-clang@10.0.0.cmake (hunk 1 above)
Comment (white238) [reply host-configs/docker/docker-linux-ubuntu18.04-x86_64-clang@10.0.0.cmake]: https://github.com/LLNL/serac/blob/12d96ac522d0a9d230df8347ac4405a0d0eca0c4/scripts/spack/configs/config.yaml#L22

This is the specific line that removes them.  Did the config.yaml not get copied somehow?  Did spack regress?

Comment (white238) [reply host-configs/docker/docker-linux-ubuntu18.04-x86_64-clang@10.0.0.cmake]: You didn't bump the spack sha... did uberenv regress and I didn't notice in my last update of it?

**Review by white238 — APPROVED** [review]

**Comment by joshessman-llnl** [comment]
LGTM

**Pull request merged**